tee-worker/litentry/core/data-providers/src/lib.rs

```diff
@@ -350,35 +343,45 @@ impl DataProviderConfig {
 		if let Ok(v) = env::var("MORALIS_API_KEY") {
 			config.set_moralis_api_key(v);
 		}
-		config
+		Ok(config)
 	}
-	pub fn set_twitter_official_url(&mut self, v: String) {
+	pub fn set_twitter_official_url(&mut self, v: String) -> Result<(), Error> {
+		check_url(&v)?;
 		debug!("set_twitter_official_url: {:?}", v);
 		self.twitter_official_url = v;
```

Comment:
Have you considered changing `twitter_official_url` type to `Url` ?

Reply:
Had similar comment above :D